examples/nlp/text_classification_with_transformer.py

```diff
@@ -0,0 +1,184 @@
+"""
+Title: Text classification with Transformer
+Author: [Apoorv Nandan](https://twitter.com/NandanApoorv)
+Date created: 2020/05/10
+Last modified: 2020/05/10
+Description: Implement transformer block as a Keras layer and use it for text classification.
+"""
+"""
+## Setup
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+
+"""
+## Implement multi head self attention as a Keras layer
+"""
+
+
+class MultiHeadSelfAttention(keras.layers.Layer):
+    def __init__(self, embed_dim, num_heads=8):
+        super(MultiHeadSelfAttention, self).__init__()
+        self.embed_dim = embed_dim
+        self.num_heads = num_heads
+        assert (
```

Comment:
Do not validate user-provided arguments with `assert`, instead, raise `ValueError`. In the message, print what was provided and what was expected.